### PR TITLE
added tests for BookController, CategoryController, BookServiceImpl, …

### PR DESCRIPTION
**Pull request by 163onmyneck**
image
image
image

### CODE REVIEW

File: src/main/java/com/example/demo/controller/CategoryController.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 @RestController
 @RequestMapping(value = "/categories")
 public class CategoryController {
+
```
Comment (MentorMateAcademy) [inline src/main/java/com/example/demo/controller/CategoryController.java]: Why do you need this line?

File: src/main/java/com/example/demo/service/book/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -60,15 +64,16 @@ public void deleteById(Long id) {
     public List<BookDto> search(BookSearchParameters bookSearchParameters, Pageable pageable) {
         Specification<Book> build = specificationBuilder.build(bookSearchParameters);
         return bookRepository.findAll(build, pageable)
-            .stream()
-            .map(bookMapper::toDto)
-            .toList();
+                             .stream()
+                             .map(bookMapper::toDto)
+                             .toList();
```
Comment (MentorMateAcademy) [inline src/main/java/com/example/demo/service/book/impl/BookServiceImpl.java]: fix your tabulation

File: src/test/java/com/example/demo/controller/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                                  .setCategories(new HashSet<>()));
+        expected.add(new BookDto().setId(2L)
+                                  .setTitle("Book Title2")
+                                  .setAuthor("Author Name2")
+                                  .setIsbn("12345678901232")
+                                  .setPrice(BigDecimal.valueOf(45.99))
+                                  .setDescription("Book description2")
+                                  .setCoverImage("image_url2")
+                                  .setCategories(new HashSet<>()));
+
```
Comment (MentorMateAcademy) [inline src/test/java/com/example/demo/controller/BookControllerTest.java]: tabulation

File: src/test/java/com/example/demo/controller/BookControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    @Transactional
+    void getBookById_ValidId_ShouldReturnBookDto() throws Exception {
+        BookDto expected = new BookDto().setId(2L)
+                                        .setTitle("Book Title2")
+                                        .setAuthor("Author Name2")
+                                        .setIsbn("12345678901232")
+                                        .setPrice(BigDecimal.valueOf(45.99))
+                                        .setDescription("Book description2")
+                                        .setCoverImage("image_url2")
+                                        .setCategories(new HashSet<>());
```
Comment (MentorMateAcademy) [inline src/test/java/com/example/demo/controller/BookControllerTest.java:125]: if you use same books lets create method that build this entity and properties better save to constants and reuse it

File: src/test/java/com/example/demo/service/category/impl/CategoryServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CategoryServiceImplTest {
+
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
```
Comment (MentorMateAcademy) [inline src/test/java/com/example/demo/service/category/impl/CategoryServiceImplTest.java]: Place it after mocks

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]
make refactoring of your tests

**Review by okuzan — APPROVED** [review]

**Pull request merged**